ACKategories/UIColorExtensions.swift

```diff
@@ -27,6 +27,21 @@ public extension UIColor {
         scanner.scanHexInt32(&rgbValue)
         self.init(hex: rgbValue)
     }
+    
+    /**
+     Returns color as hex string (eg. '#ff00ff').
+     Monochrome check included (works for white/black/clear).
+     */
+    var hexString: String {
+        let components = cgColor.components
+        let isMonochrome = cgColor.colorSpace?.model == .monochrome
+        
+        let r: CGFloat = isMonochrome ? components![0] : components![0]
```

Comment:
Not sure about this `!` it seems to work with e.g. HSB colors, but if I use `UIColor(patternImage:)`, it crashes here. Therefore I think that this property should therefore return `String?`

Reply:
I agree that we should get rid of those exclamation marks and that the return type should be `String?`. `CGColor` also have a property named `numberOfComponents`. Would it be possible to use this property in the code?

Reply:
I added `numberOfComponents` to check for colors that comes from `UIColor(patternImage:)` init, cause these colors comes with one component only(containing 0 or 1). So they cannot be trasformed to usual colors.